DNN Platform/Modules/DDRMenu/MenuNode.cs

```diff
@@ -326,6 +326,31 @@ public MenuNode FindByNameOrId(string tabNameOrId)
             return null;
         }
 
+        /// <summary>
+        /// Finds all menu nodes in the hierarchy with matching name or id.
+        /// </summary>
+        /// <param name="tabNameOrId">A string representing either the tab (page) name or id.</param>
+        /// <returns>
+        /// A <see cref="IEnumerable{MenuNode}"/> instance containing all menu nodes found.
+        /// </returns>
+        public IEnumerable<MenuNode> FindAllByNameOrId(string tabNameOrId)
+        {
+            var found = new List<MenuNode>();
+
+            foreach (var child in this.Children)
+            {
+                found.AddRange(child.FindAllByNameOrId(tabNameOrId));
+            }
+
+            if (tabNameOrId.Equals(this.Text, StringComparison.InvariantCultureIgnoreCase)
+                || tabNameOrId == this.TabId.ToString())
+            {
+                found.Add(this);
+            }
+
+            return found;

```

Comment:
I think this fix works, thanks! What do you think about making this an iterator method instead of creating a bunch of (mostly empty) lists?

```suggestion
            foreach (var found in this.Children.SelectMany(child => child.FindAllByNameOrId(tabNameOrId)))
            {
                yield return found;
            }

            if (tabNameOrId.Equals(this.Text, StringComparison.InvariantCultureIgnoreCase)
                || tabNameOrId == this.TabId.ToString())
            {
                yield return this;
            }
```

Reply:
That's of course a better solution. I was too lazy and just copied the original function. Accepting.